docs/cybersecurity/ddos/index.md

```diff
@@ -0,0 +1,217 @@
+--8<-- "docs/cybersecurity/ddos/source.json"
+```
+
+--8<-- "docs/shared/manual_insert.md"
+
+### ksqlDB code
+
+This application takes the raw network packet data and creates a structured stream of events that can be processed using SQL. Using [Windows](https://docs.ksqldb.io/en/latest/concepts/time-and-windows-in-ksqldb-queries/#windows-in-sql-queries) and filters, the application detects a high number of connection `RESET` events from the server and isolates the potentially offending source.
+
+--8<-- "docs/shared/ksqlb_processing_intro.md"
+
+``` sql
+--8<-- "docs/cybersecurity/ddos/process.sql"
+```
+
+--8<-- "docs/shared/manual_cue.md"
+
+```sql
+--8<-- "docs/cybersecurity/ddos/manual.sql"
+```
+## Cleanup
+
+--8<-- "docs/shared/cleanup.md"
+
+## Explanation
+
+This solution uses ksqlDB's ability to [model](https://www.confluent.io/blog/ksqldb-techniques-that-make-stream-processing-easier-than-ever/) and [query](https://docs.ksqldb.io/en/latest/how-to-guides/query-structured-data/) structured data. 
+
+### Streaming JSON
+
+Let's break down commands in this application and explain the individual parts.
+
+The first step is to model the packet capture data using ksqlDB's `CREATE STREAM` statement, giving our new stream the name `network_traffic`:
+
+```sql
+CREATE STREAM network_traffic
+```
+
+We then define the schema for events in the topic by declaring field names and data types using standard SQL syntax. Looking at this snippet from the full statement:
+
+```sql
+  timestamp BIGINT,
+  layers STRUCT<
+   ...
+   ip STRUCT< 
+      src VARCHAR, 
+      src_host VARCHAR, 
+      dst VARCHAR, 
+      dst_host VARCHAR, 
+      proto VARCHAR >,
+   ...
+```
+
+We are declaring an event structure that contains a timestamp field and then a child nested data structure named `layers`. Comparing the sample packet capture event with the declared structure, we see the relationships between the data and the field names and types:
+
+```json
+{
+  "timestamp": "1590682723239",
+  "layers": {
+    ...
+    "ip": {
+      "src": "192.168.33.11",
+      "src_host": "192.168.33.11",
+      "dst": "192.168.33.77",
+      "dst_host": "192.168.33.77"
+    },
+    ...
+  }
+}
+
+```
+
+The `CREATE STREAM ... WITH ...` [statement](https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/create-stream/) marries together the schema for the events with the Kafka topic. The `WITH` clause in the statement allows you to specify details about the stream. 
+
+```sql
+WITH (
+  KAFKA_TOPIC='network-traffic', 
+  TIMESTAMP='timestamp', 
+  VALUE_FORMAT='JSON', 
+  PARTITIONS=6
+);
+```
+
+The `KAFKA_TOPIC` property is required, and indicates the topic that will back the stream. The topic must either exist in Kafka, or the `PARTITIONS` property must be provided as well. If you're creating a stream for a topic that already exists, like would be the case when using a connector to source event data, you should remove the `PARTITIONS` property from this command.
```

Comment:
@rspurgeon This is a great explanation!  Since it applies to ALL recipes, WDYT if you move this into a `docs/shared/<tbd>` file and reference it that way?  Maybe in `docs/shared/ksqlb_processing_intro.md`?

Reply:
Looking at where this information is most useful, I felt it was best left independent of the note about CC console in the `ksqldb_processing_intro` file.  Instead, I put this into a shared file, `ksqldb_with_partitions_info.md` and referenced it here. This way it could be referenced from any recipe in the place that makes the most sense to that particular recipe.  LMK what you think.